inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,457 @@
+        std::vector<Type> biomass; /*!< Derived quantity: total population biomass in each year*/
+        std::vector<Type> spawning_biomass; /*!< Derived quantity: Spawning_biomass*/
+        std::vector<Type> unfished_spawning_biomass; /*!< Derived quanity Spawning biomass assuming unfished*/
+        std::vector<Type> proportion_mature_at_age; /*!< Derived quantity: Proportion matura at age */
+        std::vector<Type> expected_numbers_at_age; /*!< Expected values: Numbers at age (thousands?? millions??) */
+        std::vector<Type> catch_numbers_at_age; /*!< Expected values???: Catch in numbers at age*/
+        std::vector<Type> catch_weight_at_age; /*!< Expected values???: Weight at age for catch*/
+        std::vector<Type> expected_catch; /*!< Expected values: Catch*/
+        std::vector<Type> expected_index; /*!< Expected values: Index (CPUE)*/
+
+
+        ///recruitment
+        int recruitment_id = -999; /*!< id of recruitment model object*/
+        std::shared_ptr<fims::RecruitmentBase<Type> > recruitment; /*!< shared pointer to recruitment module */
+
+        //growth
+        int growth_id = -999; /*!< id of growth model object*/
+        std::shared_ptr<fims::GrowthBase<Type> > growth; /*!< shared pointer to growth module */
+
+        //maturity
+        int maturity_id = -999; /*!< id of maturity model object*/
+        std::shared_ptr<fims::MaturityBase<Type> > maturity; /*!< shared pointer to maturity module */
+
+        //fleet
+        int fleet_id = -999; /*!< id of fleet model object*/
+        std::vector<std::shared_ptr<fims::Fleet<Type> > > fleets; /*!< shared pointer to fleet module */
+
+        //this -> means you're referring to a class member (member of self)
+        
+        //std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+        Population() {
+            this->id = Population::id_g++;
+        }
+
+        /**
+         * @brief Initialize values. Called once at the start of model run.
+         * 
+         * @param index_ya dimension folded index for year and age
+         * @param age age index
+         */
+        void Initialize(int nyears, int nseasons, int nages) {
+            //size all the vectors to length of nages
+            nfleets = fleets.size();
+            ages.resize(nages);
+            catch_at_age.resize(nages);
+            catch_numbers_at_age.resize((nyears+1)*nages*nfleets);
+            mortality_F.resize(nyears*nages);
+            mortality_Z.resize(nages);
+            proportion_mature_at_age.resize((nyears+1)*nages);
+            weight_at_age.resize(nages);
+            catch_weight_at_age.resize(nyears*nages*nfleets);
+            unfished_numbers_at_age.resize((nyears+1)*nages);
+            numbers_at_age.resize((nyears+1)*nages);
+            expected_catch.resize(nyears*nfleets);
+            expected_index.resize(nyears*nfleets);
+            biomass.resize((nyears+1));
+            unfished_spawning_biomass.resize((nyears+1));
```

Comment:
@nathanvaughan-NOAA, could you clarify the meaning and dimension of unfished_spawning_biomass? Are they dynamic unfished spawning biomass? Is the size of unfished_spawning_biomass (nyears+1) or 1? Thank you!

Reply:
Yes, this is a dynamic unfished SSB in the event that a model had time-varying M, R0, of maturity function. The size is (nyears+1)*nages so that the model can calculate values from the start of the first year to the end of the last year.